Wei/Utility/DeepLinkActionHandler.swift

```diff
@@ -32,26 +32,25 @@ final class DeepLinkActionHandler: DeepLinkActionHandlerProtocol, Injectable {
             print("deeplink action: Sign", signedMessage, callbackScheme)
             
         case .signTransaction(let rawTransaction, let callbackScheme):
-            presentSignTransactionViewController(rawTransaction: rawTransaction, scheme: callbackScheme)
+            presentSignTransactionViewController(rawTransaction: rawTransaction, actionKind: .sign, scheme: callbackScheme)
             
         case .broadcastTransaction(let rawTransaction, let callbackScheme):
-            let signedTransaction = try walletManager.sign(rawTransaction: rawTransaction)
-            print("deeplink action: Broadcast Transaction", signedTransaction, callbackScheme)
+            presentSignTransactionViewController(rawTransaction: rawTransaction, actionKind: .broadcast, scheme: callbackScheme)
         }
     }
     
-    private func presentSignTransactionViewController(rawTransaction: RawTransaction, scheme: String) {
+    private func presentSignTransactionViewController(rawTransaction: RawTransaction, actionKind: SignTransactionViewModel.ActionKind, scheme: String) {
         if AppDelegate.rootViewController.presentedViewController != nil {
             AppDelegate.rootViewController.dismiss(animated: true) { [weak self] in
-                self?.presentSignTransactionViewController(rawTransaction: rawTransaction, scheme: scheme)
+                self?.presentSignTransactionViewController(rawTransaction: rawTransaction, actionKind: actionKind, scheme: scheme)
             }
         }
         
-        let viewController = SignTransactionViewController.make(rawTransaction: rawTransaction) { [weak self] signature in
+        let viewController = SignTransactionViewController.make(rawTransaction: rawTransaction, actionKind: actionKind) { [weak self] string in
             guard let url = self?.buildURL(
                 scheme: scheme,
-                path: "/sign_transaction",
-                queryItems: URLQueryItem(name: "signature", value: signature)) else {
+                path: actionKind == .sign ? "/sign_transaction" : "/broadcast_transaction",
```

Comment:
We need to add or remove if or switch statement when we add other action, so I think we should call a function which is defined in `ActionKind` enum.